pkg/operators/ebpf/ebpf.go

```diff
@@ -323,6 +333,45 @@ func (i *ebpfInstance) init(gadgetCtx operators.GadgetContext) error {
 	return nil
 }
 
+func (i *ebpfInstance) addExtraInfo(gadgetCtx operators.GadgetContext) error {
+	ef, err := elf.NewFile(bytes.NewReader(i.program))
+	if err != nil {
+		return fmt.Errorf("parsing elf file: %w", err)
+	}
+	var sections []string
+	var maps []*api.Map
+	var programs []*api.Program
```

Comment:
In addition to programs and maps, we could add variables as understood by cilium/ebpf:
https://github.com/cilium/ebpf/blob/v0.17.3/collection.go#L46